src/axom/sidre/core/View.hpp

```diff
@@ -1514,10 +1547,85 @@ class View
   State getStateId(const std::string& name) const;
 
   /*!
-   * \brief Private method. If allocatorID is a valid allocator ID then return
-   *  it. Otherwise return the ID of the default allocator of the owning group.
+   * \brief Private method. If allocatorID is a valid Axom allocator ID then return
+   *  it. Otherwise return the ID of the default Axom allocator of the owning group.
    */
-  int getValidAllocatorID(int allocatorID);
+  int getValidAxomAllocatorID(int allocatorID);
+
+  /*!
+   * \brief Private method. If allocatorID is a valid Axom allocator
+   *  ID then return its corresponding Conduit allocator ID. Otherwise
+   *  return the corresponding Conduit allocator ID of the default
+   *  allocator of the owning group.
+   */
+  int getValidConduitAllocatorID(int allocatorID);
+
+  /*!
+   * \brief Return whether view data is accessible on the host CPU,
+   * as determined by Axom's memory management.
+   *
+   * If data is external and not allocated through Umpire or malloc,
+   * the behavior is undefined.
+   *
+   * By convention, this returns a false if data is not allocated,
+   * because we expect null pointers to be correctly checked before use.
+   */
+  bool isHostAccessible() const;
+
+  //!@brief Print on host, as a single line.
+  template <typename T>
+  void hostPrintScalar(std::ostream& os = std::cout) const
+  {
+    if(isHostAccessible())
+    {
+      os << ' ' << T(m_node.value());
+    }
+    else
+    {
+      os << ' ' << getVoidPtr() << " # non-host " << typeid(T).name() << " data";
+    }
+  }
+
+  //!@brief Print on host, as a single line.
+  template <typename T>
+  void hostPrintArray(std::ostream& os = std::cout) const
+  {
+    if(isHostAccessible())
+    {
+      os << " [";
+      auto start = (T*)(getVoidPtr());
+      auto end = (T*)(getVoidPtr()) + getNumElements();
+      if(getNumElements() <= 10)
```

Comment:
What is significant about the numbers `10` and `5` here? What's the difference in processing one size versus the other?

Reply:
Those are arbitrary values.  If there are more than 10 items, print the first and last 5.  The middle ones are printed as `...`.  I'll create some variable names to help clarify.